src/SPC/doctor/item/LinuxMuslCheck.php

```diff
@@ -8,54 +8,77 @@
+use SPC\exception\WrongUsageException;
+use SPC\store\Downloader;
+use SPC\store\FileSystem;
 
 class LinuxMuslCheck
 {
     /** @noinspection PhpUnused */
+    /**
+     * @throws WrongUsageException
+     */
     #[AsCheckItem('if musl-libc is installed', limit_os: 'Linux')]
-    public function checkMusl(): ?CheckResult
+    public function checkMusl(): CheckResult
     {
-        $file = sprintf('/lib/ld-musl-%s.so.1', php_uname('m'));
-        if (file_exists($file)) {
+        $arch = arch2gnu(php_uname('m')) === 'x86_64' ? 'x86_64-linux-musl' : 'aarch64-linux-musl';
+        $cross_compile_lib = "/usr/local/musl/{$arch}/lib/libc.a";
+        $musl_wrapper_lib = sprintf('/lib/ld-musl-%s.so.1', php_uname('m'));
+        if (file_exists($musl_wrapper_lib) && file_exists($cross_compile_lib)) {
             return CheckResult::ok();
         }
-
-        // non-exist, need to recognize distro
-        $distro = SystemUtil::getOSRelease();
-        return match ($distro['dist']) {
-            'ubuntu', 'alpine', 'debian', 'rhel', 'almalinux' => CheckResult::fail('musl-libc is not installed on your system', 'fix-musl', [$distro]),
-            default => CheckResult::fail('musl-libc is not installed on your system'),
-        };
+        return CheckResult::fail('musl-libc is not installed on your system', 'fix-musl');
     }
 
+    /** @noinspection PhpUnused */
     /**
      * @throws RuntimeException
-     * @noinspection PhpUnused
+     * @throws DownloaderException
+     * @throws FileSystemException
+     * @throws WrongUsageException
      */
     #[AsFixItem('fix-musl')]
-    public function fixMusl(array $distro): bool
+    public function fixMusl(): bool
     {
-        $rhel_install = 'wget https://musl.libc.org/releases/musl-1.2.4.tar.gz && tar -zxvf musl-1.2.4.tar.gz && \
-                         rm -f musl-1.2.4.tar.gz && cd musl-1.2.4 && 
-                         if [[ ! "$PATH" =~ (^|:)"/usr/local/musl/bin"(:|$) ]]; then echo "export PATH=/usr/local/musl/bin:$PATH" >> ~/.bash_profile
-                         fi && \
+        $musl_source = [
+            'type' => 'url',
+            'url' => 'https://musl.libc.org/releases/musl-1.2.4.tar.gz',
+        ];
+        $musl_cross_compile_source = [
+            'type' => 'git',
+            'url' => 'https://github.com/richfelker/musl-cross-make',
+            'rev' => 'master',
+        ];
+        Downloader::downloadSource('musl-cross-make', $musl_cross_compile_source);
+        Downloader::downloadSource('musl-1.2.4', $musl_source);
+        FileSystem::extractSource('musl-1.2.4', DOWNLOAD_PATH . '/musl-1.2.4.tar.gz');
+        $arch = arch2gnu(php_uname('m')) === 'x86_64' ? 'x86_64-linux-musl' : 'aarch64-linux-musl';
+        $install_musl_wrapper_cmd = 'cd ' . DOWNLOAD_PATH . '/musl-cross-make && \
+                       make install TARGET=' . $arch . ' OUTPUT=/usr/local/musl -j';
+        $musl_install = 'cd ' . SOURCE_PATH . '/musl-1.2.4 && \
```

Comment:
we need to install both, musl-cross-make for c++ support, musl-gcc to be able to run dynamically linked binaries built with musl-cross-make